mistletoe/html_attributes_renderer.py

```diff
@@ -0,0 +1,201 @@
+"""
+HTML Attributes renderer for mistletoe.
+"""
+
+import html
+from mistletoe import block_token
+from mistletoe import span_token
+from mistletoe.block_token import HTMLAttributes
+from mistletoe.html_renderer import HTMLRenderer
+
+
+class HTMLAttributesRenderer(HTMLRenderer):
+    """
+    HTML Attributes renderer class.
+
+    See mistletoe.html_renderer module for more info.
+    """
+    def __init__(self):
+        """
+        Args:
+            extras (list): allows subclasses to add even more custom tokens.
+        """
+        super().__init__(HTMLAttributes)
+        self.RENDERER_START = False
+
+    def render(self, token):
+        """
+        Grabs the class name from input token and finds its corresponding
+        render function.
+
+        Basically a janky way to do polymorphism.
+
+        Arguments:
+            token: whose __class__.__name__ is in self.render_map.
+        """
+        # reconcile our htmlattributes
+        if not self.RENDERER_START: 
+            self.reconcile_attrs(token)
+        return self.render_map[token.__class__.__name__](token)
+
+    def reconcile_attrs(self, doc_token):
+        """Traverse token children while assigning html attributes if available"""
+        self.RENDERER_START = True
+        recon_tokens = []
+        htmlAttributesToken: block_token.HTMLAttributes = None
+        for token_type in doc_token.children:
+            if 'HTMLAttributes' == token_type.__class__.__name__: 
+                htmlAttributesToken = token_type
+                continue
+            if htmlAttributesToken: 
+                htmlAttributesToken.apply_props(token_type)
+                htmlAttributesToken.clear()
+                htmlAttributesToken = None
+            recon_tokens.append(token_type)
+        doc_token.children = recon_tokens
+
+    def render_strong(self, token: span_token.Strong) -> str:
```

Comment:
This looks identical to the base class implementation (HTMLRenderer). Maybe I'm missing something, but if that's the case then I'd suggest to remove it from here.

Reply:
See (a7828be24ef1dba1cad7426d25561286c3be95e1)

Reply:
OK. Leaving this thread opened, so that we don't forget there are still some copied methods to be altered.

BTW when thinking about it, this methods' duplication is not ideal. Because when we change something in the `HtmlRenderer`, we should most probably also apply the same to the new `HtmlAttributesRenderer`, right? So putting this new feature introduced by this PR into the `HtmlRenderer` _directly_ should be our final goal, shouldn't it?